kotlinpoet/src/jvmTest/kotlin/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -4416,6 +4416,47 @@ class TypeSpecTest {
 
     assertThat(toString(type)).isEqualTo(expect)
   }
+// https://github.com/square/kotlinpoet/issues/2033
+  @Test fun testDelegateOnAnonymousObject() {
+    val type = TypeSpec.anonymousClassBuilder()
+      .addSuperinterface(
+        Consumer::class.parameterizedBy(String::class),
+        CodeBlock.of("java.util.function.Consumer({ println(it) })"),
+      )
+      .build()
+
+    val expect = """
+        |object : java.util.function.Consumer<kotlin.String> by java.util.function.Consumer({ println(it) }) {
+        |}
+    """.trimMargin()
+
+    assertThat(type.toString()).isEqualTo(expect)
+  }
+  // https://github.com/square/kotlinpoet/issues/2033
+  @Test fun testMultipleDelegatesOnAnonymousObject() {
+    val type = TypeSpec.anonymousClassBuilder()
+      .primaryConstructor(
+        FunSpec.constructorBuilder()
+          .build(),
+      )
```

Comment:
nit: the constructor is ignored, right? Or does removing it actually modify the output?

Reply:
You're correct. This was copy pasta from the `testMultipleDelegates` test I based mine off of